File: SSI_2nd_year/SSI-agent/SSI_Shop_Server_Running/src/main/java/com/bd/ssi/common/api/ApiResponse.java
package com.bd.ssi.common.api;

/**
 * API 응답 클래스
 */
public class ApiResponse {
    public static class CODE {
        public static final int SUCCESS = 1000;
        public static final int FAIL_SERVER_ERROR = 9999;
    }

    private Integer code;
    private Object data;
    private String message;

    public ApiResponse(Integer code, Object data, String message) {
        this.code = code;
        this.data = data;
        this.message = message;
    }

    public static ApiResponse ok(Object data) {
        return new ApiResponse(CODE.SUCCESS, data, null);
    }
    public static ApiResponse fail(Integer code, String message){
        return new ApiResponse(code, null, message);
    }

    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: SSI_2nd_year/SSI-app/SSI_Shop_Android_Running/app/src/main/java/com/bd/ssishop/api/ApiResponse.kt
package com.bd.ssishop.api

/**
 * Api 공통 응답 객체 생성
 */
data class ApiResponse<T>(
    val code: Int,
    val data: T? = null,
    val message: String? = null
) {
    companion object {
        inline fun <reified T> error(code: Int, message: String? = null) = ApiResponse(code, null as T?, message )
    }
}